### ISSUE TITLE
Implement CSS caret-color

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/UXP#1668]
Something apparently used in some frameworks as part of an illogical cascade for input styling, our missing support for the CSS3 `caret-color` property is causing some web compatibility and accessibility issues (e.g. invisible input cursor).

This should be trivial to port from Mozilla (Firefox 53 implementation)
See [BZ 1063162](https://bugzilla.mozilla.org/show_bug.cgi?id=1063162)

**Comment by athenian200** [comment]
I actually have an implementation of this that I worked on a while ago and forgot to open an issue for it, because I decided to stop work on it to focus on the opacity thing.

There is one problem with it, though. There's a test that involves making sure the caret is visible when you have grey text on a grey background. That test fails and it's apparently almost impossible to detect that condition. Whatever mechanisms we have for ensuring sufficient contrast already work just well enough to make the two shades of grey slightly different so that the similarity can't be detected using a simple equality check. I looked into what it would take to check whether two colors are similar enough that the caret can't be seen, and there doesn't seem to be a way of doing it that's not a computationally expensive nightmare. 

If we didn't have a caret-color implementation to worry about, we could just check luminosity and force the caret to be black or white depending on which would be more visible against the given background, but since we do have to consider colors, all kinds of crazy stuff like CIE and LAB76 that's computationally expensive would come into play.

The good news is that I just checked and found that apparently Firefox and Chrome fail this test too, so unless we really care about making the input caret visible when the text and background are the same color for accessibility reasons, it's done.

**Comment by wolfbeast** [comment]
If you're talking about the whole "inverted" bit, everyone decided not to make that part of the spec.
